src/utils.h

```diff
@@ -21,6 +21,60 @@
 #include "families.h"
 #include "math.h"
 
+inline void PredictSample(std::vector<double>&       prediction,
```

Comment:
Consider a different name - this calculates the _linear predictor_, but it isn't necessarily a prediction for a given GLM. (I.e., in logistic land, this doesn't give `y_hat` - it gives `InvLogit(y_hat)`)

Reply:
Right, you preferred ConditionalMean didn't you?

Reply:
Not here if I'm reading the code correctly - this is the linear predictor (X*beta + intercept + offset). The conditional mean is obtained by pushing the linear predictor through the link function.